### PR TITLE
Migrate netapi tests to PyTest

### PR DESCRIPTION
**Pull request by s0undt3ch**
### What does this PR do?
See title
This is also the precursor for the fix to #56495 on https://github.com/saltstack/salt/pull/60493

### CODE REVIEW

**Review by Ch3LL — DISMISSED** [review]

File: tests/pytests/integration/netapi/test_ssh_client.py [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+
+def test_shell_inject_ssh_priv(
+    client, salt_ssh_roster_file, rosters_dir, tmp_path, salt_auto_account
+):
+    """
+    Verify CVE-2020-16846 for ssh_priv variable
+    """
+    # ZDI-CAN-11143
+    path = tmp_path / "test-11143"
+    tgt = "www.zerodayinitiative.com"
```
Comment (Ch3LL) [inline tests/pytests/integration/netapi/test_ssh_client.py:127]: How about we use https://pytest-httpserver.readthedocs.io/en/latest/ here or something similar, so we are not depending on external resources.

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: tests/pytests/integration/netapi/test_ssh_client.py (hunk 1 above)
Comment (s0undt3ch) [reply tests/pytests/integration/netapi/test_ssh_client.py:127]: Yep, that might be a good approach going forward.

Comment (Ch3LL) [reply tests/pytests/integration/netapi/test_ssh_client.py]: Did you want to handle that in this PR or a follow up one? I'm fine either way.

Comment (s0undt3ch) [reply tests/pytests/integration/netapi/test_ssh_client.py]: This one still needs work, i have to revisit this.
But probably on a separate PR, where we address all webservers we use ...

**Comment by Ch3LL** [comment]
Are there still plans to try to get this in for phosphorus or can I take off the phosphorus label?

**Comment by s0undt3ch** [comment]
Switched to 3006 and converted to draft

**Comment by s0undt3ch** [comment]
re-run full all

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**